.github/workflows/validate.yaml

```diff
@@ -0,0 +1,46 @@
+name: validate-files
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    # Allows you to run this workflow manually
+  workflow_dispatch:
+
+jobs:
+  validate-files:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
+        with:
+          path: |
+            **/node_modules
+            .yarn/cache
+          key: ${{ runner.os }}-yarn-cache-${{ hashFiles('./yarn.lock') }}
+
+      - name: yarn-install
+        run: |
+          yarn install
+          CHANGES=$(git status -s)
+          if [[ ! -z $CHANGES ]]; then
+            echo "Changes found: $CHANGES"
+            git diff
+            exit 1
+          fi
+
+      - name: setup-node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20
```

Comment:
Just curious, what happens if you remove this? I think the ubuntu image already has node. That's why the CI job doesn't include this step and why the `yarn install` step before it works.

Reply:
So doing a bit more digging, apparently the reason people still include this is that they don't want to rely on the default that comes installed with `ubuntu-latest` and have a bit more of a controlled environment. Other than that it also manages the Node.js installation in a way that is more optimized for Github Actions.

Reply:
I understand but we should be consistent. Either use it in all the workflows or none. And if you're going to add it, use `actions/setup-node@v4` which is the latest.

Reply:
Agreed here, I will remove this part for now and then look to add it back to every workflow after further discussions here

Reply:
Agreed here, I will look into adding it to all the other workflows on another PR, we can discuss this further